### PR TITLE
Upgrade `rz-test` CI to `ubuntu-22.04`

### PR DESCRIPTION
**Pull request by kazarmy**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This pr upgrades the Linux `rz-test` CI to `ubuntu-22.04` and is a continuation of #3062. Similar considerations apply:

1. Ubuntu 22.04 is already at its first point release (22.04.1) so this upgrade should be safe (though the debug tests might be another matter).
2. `ubuntu-latest` [is still on `ubuntu-20.04`](https://github.com/actions/runner-images/tree/2aec524970584138c5986109ca7be7ae89b0899d#available-images) and I think it's better to fix the ubuntu version so that it doesn't change unexpectedly.

**Test plan**

All builds are green.

**Closing issues**

...

### CODE REVIEW

**Review by ret2libc — APPROVED** [review]

**Comment by XVilka** [comment]
What about using gcc-12 with ASAN builds instead?

**Comment by kazarmy** [comment]
sgtm faa48c33d8efc0328e4deffcb1c3c837a95f51b6

**Comment by XVilka** [comment]
@kazarmy please rebase

File: librz/asm/arch/arc/gnu/arcompact-dis.c [hunk 1]
```diff
@@ -3731,7 +3731,7 @@ int ARCompact_decodeInstr(bfd_vma address, /* Address of this instruction.  */
 	fprintf_ftype func = info->fprintf_func;
 	int bytes;
 	int lowbyte, highbyte;
-	char buf[256];
+	char buf[allOperandsSize + 1];
```
Comment (wargio) [inline librz/asm/arch/arc/gnu/arcompact-dis.c]: some compilers does not like this. use malloc

Comment (kazarmy) [reply librz/asm/arch/arc/gnu/arcompact-dis.c]: `allOperandsSize` is an enum constant so there shouldn't be a problem

Comment (wargio) [reply librz/asm/arch/arc/gnu/arcompact-dis.c]: oh, then yeah. sorry

Comment (kazarmy) [reply librz/asm/arch/arc/gnu/arcompact-dis.c]: if this was rizin code, it would be in all caps

**Comment by XVilka** [comment]
Once GCC 12 is supported on ASAN builds, I also recommend enabling `-ftrivial-auto-var-init=pattern` flag, which will fill all uninitialized memory (variables, structures, etc) with 0xFE, which will likely trigger bugs if we have some uninitialized places.

**Comment by kazarmy** [comment]
Sgtm

**Comment by wargio** [comment]
i agree. that will break rizin a lot

**Comment by XVilka** [comment]
```
FAILED: librz/core/librz_core.so.0.5.0.p/cmd_cmd.c.o 
gcc-12 -Ilibrz/core/librz_core.so.0.5.0.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -Ilibrz/core/cmd_descs -I../librz/core/cmd_descs -Ilibrz/util/sdb/src -I../librz/util/sdb/src -Ilibrz/util/sdb -I../librz/util/sdb -I../librz/asm/arch/include -I../librz/asm/arch -I../librz/asm/arch/h8300 -I../librz/asm/arch/hexagon -I../librz/asm/arch/msp430 -I../librz/asm/arch/rsp -I../librz/asm/arch/mcore -I../librz/asm/arch/v850 -I../librz/asm/arch/propeller -I../librz/asm/arch/ebc -I../librz/asm/arch/cr16 -I../librz/asm/arch/8051 -I../librz/asm/arch/v810 -I../librz/asm/arch/or1k -I../librz/analysis/arch/gb -I../librz/type/parser -I../librz/bin/format/elf -I../librz/bin/format/dmp -I../librz/bin/format/mdmp -I../librz/bin/format/pe -I../librz/bin/format -I../subprojects/rzgdb/include -I../subprojects/rzgdb/include/gdbclient -I../subprojects/rzgdb/include/gdbserver -Isubprojects/rzheap -I../subprojects/rzheap -I../subprojects/rzheap/rz_jemalloc -I../subprojects/rzheap/rz_windows -I../subprojects/rizin-shell-parser/src/tree_sitter -I../subprojects/tree-sitter-0.20.6/lib/src -I../subprojects/tree-sitter-0.20.6/lib/include -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O2 -g -Wimplicit-fallthrough=3 -DRZ_PLUGIN_INCORE=1 -D_GNU_SOURCE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -DASAN=1 -DRZ_ASSERT_STDOUT=1 -Wno-cpp -fPIC -MD -MQ librz/core/librz_core.so.0.5.0.p/cmd_cmd.c.o -MF librz/core/librz_core.so.0.5.0.p/cmd_cmd.c.o.d -o librz/core/librz_core.so.0.5.0.p/cmd_cmd.c.o -c ../librz/core/cmd/cmd.c
In file included from ../librz/core/cmd/cmd.c:79:
../librz/core/cmd/cmd_search.c: In function ‘do_analysis_search’:
../librz/core/cmd/cmd_search.c:2054:62: error: ‘strcmp’ reading 1 or more bytes from a region of size 0 [-Werror=stringop-overread]
 2054 |                                                         if (!strcmp(input + 1, type)) {
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/string.h:535,
                 from ../librz/include/rz_types.h:202,
                 from ../librz/include/rz_main.h:8,
                 from ../librz/include/rz_core.h:7,
                 from ../librz/core/cmd/cmd.c:7:
In function ‘memmove’,
    inlined from ‘rz_core_cmd_subst’ at ../librz/core/cmd/cmd.c:1205:3:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:36:10: error: ‘__builtin_memmove’ offset 0 is out of the bounds [0, 0] [-Werror=array-bounds]
   36 |   return __builtin___memmove_chk (__dest, __src, __len,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   37 |                                   __glibc_objsize0 (__dest));
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**Comment by kazarmy** [comment]
Will get to them soonish

**Review by XVilka — APPROVED** [review]
Finally, everything except the static build (https://github.com/rizinorg/rizin/issues/3220 - and it was happening even before this PR) looks green. Great job!

**Comment by kazarmy** [comment]
Thanks! Ok merging...

**Pull request merged**